### PR TITLE
[BugFix] Fix Struct<ObjectColumn> checksum crash

### PR DESCRIPTION
**Pull request by wyb**
## Why I'm doing:
ASAN BE crashes on checksum task because `ObjectColumn::xor_checksum` is not supported.

## What I'm doing:
checksum task ignores the column which type does not support checksum or sub columns do not support checksum.

Fixes #issue
https://github.com/StarRocks/StarRocksTest/issues/7258

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/task/engine_checksum_task.cpp [hunk 1]
```diff
@@ -77,18 +77,8 @@ Status EngineChecksumTask::_compute_checksum() {
         return Status::OK();
     }
 
-    std::vector<uint32_t> return_columns;
     auto tablet_schema = tablet->tablet_schema();
-
-    size_t num_columns = tablet_schema->num_columns();
-    for (size_t i = 0; i < num_columns; ++i) {
-        LogicalType type = tablet_schema->column(i).type();
-        if (is_support_checksum_type(type)) {
```
Comment (trueeyu) [inline be/src/storage/task/engine_checksum_task.cpp:86]:
```
inline bool is_support_checksum_type(LogicalType type) {
    return type != TYPE_FLOAT && type != TYPE_DOUBLE && type != TYPE_HLL && type != TYPE_OBJECT &&
           type != TYPE_PERCENTILE && type != TYPE_JSON;
}
```

Why reach the `ObjectColumn<T>::xor_checksum`?

File: be/src/column/object_column.cpp [hunk 2]
```diff
@@ -260,7 +260,6 @@ void ObjectColumn<T>::crc32_hash(uint32_t* hash, uint32_t from, uint32_t to) con
 
 template <typename T>
 int64_t ObjectColumn<T>::xor_checksum(uint32_t from, uint32_t to) const {
-    DCHECK(false) << "object column shouldn't call xor_checksum";
```
Comment (trueeyu) [inline be/src/column/object_column.cpp:263]: This change will produce unnecessary scan, for bitmap and other columns ?

Comment (trueeyu) [reply be/src/column/object_column.cpp:263]: Better use schema to check.

File: be/src/storage/task/engine_checksum_task.cpp (hunk 1 above)
Comment (wyb) [reply be/src/storage/task/engine_checksum_task.cpp:86]: use struct<a bitmap, b bitmap> type in the test case

**Review by Seaven — DISMISSED** [review]

File: be/src/column/object_column.cpp (hunk 2 above)
Comment (wyb) [reply be/src/column/object_column.cpp]: done

**Review by trueeyu — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by wyb** [comment]
https://github.com/Mergifyio backport branch-3.2

**Comment by wyb** [comment]
https://github.com/Mergifyio backport branch-3.1

**Pull request merged**